### PR TITLE
REG-568: implemented configuring reverse resolution automatically

### PR DESCRIPTION
**Pull request by nickshatilo**
### Background

In order to improve usage of our reverse resolution, we want to configure it by default while minting domains. This PR introduces this functionality.

### Changes

* Added `_safeSetReverse` call into `_mint` and `_safeMint` functions, so that mints with URI (from `MintingManager`) will all be triggering installing of reverse resolution if it's not already set
* Handled the case when minting TLDs, so that we don't set reverse resolution for `0xdead` address
* Covered changes by specs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## v0.4.9
+- Added configuring reverse resolution on minting automatically
+
+## v0.4.8
+- Added YLD token to resovler list
```
Comment (nickshatilo) [inline CHANGELOG.md:5]: Added this one cause was missing in the previous (https://github.com/unstoppabledomains/uns/pull/123) PR

**Review by aquiladev — COMMENTED** [review]
I'm curious about gas usage difference. What delta for UNSRegistry size and delta for gas usage of domain minting?

**Comment by aquiladev** [comment]
gas usage delta looks too small, emit event costs ~3k gas

**Comment by nickshatilo** [comment]
Hmm, yeah. Rechecked it, the thing is that in the gas tests it runs 3 mints and compares gas prices with forwarder and direct ones. And as we set reverse only once per user / network, the only mint that could be compared is the first execute one.

So the current execute of `mintSLD` runs for `165.5k` of gas, and now it's `189.4k` of gas. So actually the delta is bigger, `~13%`

**Comment by nickshatilo** [comment]
@aquiladev I added removing of Reverse Resolution into the gas spec, so here are the accurate values:

Before:

```
─────────────────────────────────────────┬─────────┬──────────┬──────────┬───────────┐
│ (index) │                  selector                   │ records │   send   │ execute  │ increase  │
├─────────┼─────────────────────────────────────────────┼─────────┼──────────┼──────────┼───────────┤
│    0    │      'mintSLD(address,uint256,string)'      │   '-'   │ '105771' │ '165424' │ '56.40 %' │
│    1    │    'safeMintSLD(address,uint256,string)'    │   '-'   │ '109527' │ '152109' │ '38.88 %' │
│    2    │ 'safeMintSLD(address,uint256,string,bytes)' │   '-'   │ '110131' │ '153031' │ '38.95 %' │
└─────────┴─────────────────────────────────────────────┴─────────┴──────────┴──────────┴───────────┘
```

After:

```
┌─────────┬─────────────────────────────────────────────┬─────────┬──────────┬──────────┬───────────┐
│ (index) │                  selector                   │ records │   send   │ execute  │ increase  │
├─────────┼─────────────────────────────────────────────┼─────────┼──────────┼──────────┼───────────┤
│    0    │      'mintSLD(address,uint256,string)'      │   '-'   │ '129717' │ '189370' │ '45.99 %' │
│    1    │    'safeMintSLD(address,uint256,string)'    │   '-'   │ '133473' │ '176055' │ '31.90 %' │
│    2    │ 'safeMintSLD(address,uint256,string,bytes)' │   '-'   │ '134077' │ '176977' │ '32.00 %' │
└─────────┴─────────────────────────────────────────────┴─────────┴──────────┴──────────┴───────────┘
```

The difference is around `13-14%`, 24k of gas.

**Review by bogdan — APPROVED** [review]

**Comment by aquiladev** [comment]
@nickshatilo what about UNSRegistry size?

**Comment by nickshatilo** [comment]
@aquiladev Yep, in terms of size `UnsRegistry` changed from `21.20 KB` to `21.29 KB`. So the difference is `0.09 KB` ~= `92` bytes

**Review by aquiladev — APPROVED** [review]

**Pull request merged**